packages/material-ui-styled-engine-sc/src/styled.test.js

```diff
@@ -0,0 +1,20 @@
+import { expect } from 'chai';
+import styled from './index';
+
+describe('styled', () => {
+  it('should help debug wrong args', () => {
+    expect(() => {
+      expect(() => {
+        styled('span')();
+      }).to.throw("Cannot read property 'length' of undefined");
+    }).toErrorDev('Material-UI: the styled("div")(styles) API requires the style to be provided');
```

Comment:
If this throws anyway, shouldn't we just throw with a helpful error message instead? If you use `MuiError` (check the codebase for existing usages) the message is minified and has minimal bundle size impact.

Reply:
only sc throw, not emotion, which is currently a lot more popular: https://npm-stat.com/charts.html?package=@material-ui/styled-engine,@material-ui/styled-engine-sc

Reply:
Also, for performance, I think that bypassing completely the branch in production would be better.

Reply:
> only sc throw, not emotion, which is currently a lot more popular:

Then just apply it to `styled-engine-sc`. 

> Also, for performance, I think that bypassing completely the branch in production would be better.

Disregard without measurement.

Reply:
> Then just apply it to styled-engine-sc.

The value to apply it to styled-engine too is to know ahead of time that we have an issue with styled-components.

Reply:
I don't know anymore what's going on. Maybe the motivation behind the styled-components adapter changed. Maybe it wasn't implemented as originally agreed upon. The current state is not ok but I don't think you care anymore so merge if you want.

Reply:
To expand on the previous point, @material-ui/styled-engine exposes **one** interface. Per this constraint, we can only support the intersection of what em and sc can support. Here, we have two options, we need to arbitrate between supporting an empty style or not. 

What's the issue with the current state?